src/main/java/org/jeasy/states/core/FiniteStateMachineImpl.java

```diff
@@ -127,6 +128,28 @@ public State getInitialState() {
         return finalStates;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Set<State> getTargetStates() {
+        return transitions.stream()
+                .filter(it -> it.getSourceState().equals(getCurrentState()))
```

Comment:
This condition is incomplete, it should be like https://github.com/j-easy/easy-states/blob/6b72eff96cb90f966fab32245c09d56611735513/src/main/java/org/jeasy/states/core/FiniteStateMachineImpl.java#L72-L74. Otherwise it is inconsistent and incorrect.